src/Native/Runtime/CachedInterfaceDispatch.cpp

```diff
@@ -0,0 +1,551 @@
+//
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+// ==--==
+//
+// Shared (non-architecture specific) portions of a mechanism to perform interface dispatch using an alternate
+// mechanism to VSD that does not require runtime generation of code.
+//
```

Comment:
I have just added this file from the TFS side, so that we can see the delta and can merge it smoothly.